core/control-plane/contract-core/src/test/java/org/eclipse/edc/connector/contract/validation/ContractValidationServiceImplTest.java

```diff
@@ -217,14 +217,35 @@ void verifyContractAgreementExpired() {
         var isValid =
                 validateAgreementDate(MIN.getEpochSecond(), MIN.getEpochSecond(), past);
 
-        assertThat(isValid).isFalse();
+        assertThat(isValid.failed()).isTrue();
     }
 
     @Test
     void verifyContractAgreementNotStartedYet() {
         var isValid = validateAgreementDate(MIN.getEpochSecond(), MAX.getEpochSecond(), MAX.getEpochSecond());
 
-        assertThat(isValid).isFalse();
+        assertThat(isValid.failed()).isTrue();
+    }
+
+    @Test
+    void validateAgreementfailsIfIdIsNotValid() {
```

Comment:
This is a super-nit but if you use camel case, can you please capitalize to "Fails"